src/cli.rs

```diff
@@ -73,12 +65,50 @@ pub fn parse_args() -> Config {
     let dir_string = matches
         .get_one::<String>("directory")
         .unwrap_or(&default_dir);
-    let directory = dir_string
+    let mut directory = dir_string
         .replace('/', &MAIN_SEPARATOR.to_string())
         .replace('\\', &MAIN_SEPARATOR.to_string())
         .parse()
         .expect("failed to parse path");
 
+    let mut root_path = std::env::current_dir().unwrap_or_else(|e| {
+        eprintln!("Cannot find the current working directory: {}", e);
+        std::process::exit(1);
+    });
+
+    if let Some(root_dir) = matches.get_one::<String>("root-dir") {
+        let path = Path::new(
+            &root_dir
+                .replace('/', &MAIN_SEPARATOR.to_string())
+                .replace('\\', &MAIN_SEPARATOR.to_string()),
+        )
+        .to_path_buf();
+        if !path.is_dir() {
+            eprintln!("Root path {:?} must be a directory!", path);
+            std::process::exit(1);
+        }
+        root_path = path.canonicalize().unwrap();
+    }
+
+    directory = root_path
+        .join(&directory)
+        .canonicalize()
+        .unwrap_or_else(|e| {
+            eprintln!(
+                "Relative path from privided root {:?} is malformed: {:?} --- {}",
+                root_path, directory, e
+            );
+            std::process::exit(1);
+        });
```

Comment:
Here I make the decision to canonicalize (and make this a full path) from the root path.

---

It does beg the question: what is the root path in conjunction with a directory path actually for? I think all I see root used as pointing to a config file location ***and*** place to walk flies within (kinda like this pattern: https://classic.yarnpkg.com/lang/en/docs/cli/#toc-cwd ). Walking otherwise would need to be set in the CLI directory argument... Why not instead remove the notion of a "root" with the provided directory being the source of walking files, and point optionally to a config file? 

So in a config file you could set a _relative_ path that the directory argument would default to if nothing was provided on the CLI.
> This would perhaps be strange, as if the config was looking for a path that was not in the present working dir, that would fail, but at least in the present impl it would fail with an error showing the full path tried. Perhaps a warning that the config set this would be sufficient to help end users.
> In a global config, for example, you would not want to set this argument at all, only on a per project basis.

Reply:
https://github.com/becheran/mlc#usage 

> All links to the file system starting with a slash on linux or backslash on windows will use another virtual root dir. For example the link in a file `[link](/dir/other/file.md)` checked with the cli arg `--root-dir /env/another/dir` will let mlc check the existence of `/env/another/dir/dir/other/file.md`.